streaming/base/world.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2022 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Information about nodes, ranks, and workers."""
+
+from torch.utils.data import get_worker_info
+
+from streaming.base.distributed import get_local_world_size, get_rank, get_world_size
+
+
+class World:
+    """Information about the nodes, ranks and workers of this run.
+
+    Warning: be careful as to whether this object was initialized in a worker (if workers are used)
+    or in a rank (which will claim one worker per rank).
+
+    Nodes:
+      - node / num_nodes
```

Comment:
just for our own understanding, we can add a comment that `each node should have a same number of ranks (GPUs)`, meaning, we support homogeneous clusters.

Reply:
Added explanatory comments